script.js

```diff
@@ -1,4 +1,125 @@
+//user name to personalize the game
+var userName = "";
+
+//number of times player wins vs computer wins vs draws
+var mode = "";
+
+//set number of player wins, computer wins, draws to 0
+var numPlayerWins = 0;
+var numComputerWins = 0;
+var numDraws = 0;
+
+//game mode normal
+var mode = "normal";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //computer choice
+  var comChoice = generateComChoice();
+  console.log(comChoice);
+  console.log(input);
+
+  console.log(userName);
+  if (userName == "") {
+    userName = input;
+    return `${userName}, let's play!`;
+  }
+  console.log(userName);
+
+  if (input == `reversePlay`) {
+    mode = `reverse`;
+    return "reverse mode activated";
+  } else if (input == `normalPlay`) {
+    mode = `normal`;
+    return "back to normal mode";
+  }
+
+  if (mode == `reverse`) {
+    return reverseGame(input, comChoice);
+  }
+
+  return normalGame(input, comChoice);
+};
+
+//generate a random number between 0 and 3
+var generateNum = function () {
+  var randomDecimal = Math.random() * 3;
+
+  //convert to integer
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+
+//assign an element to a value between 0 and 3
+var generateComChoice = function () {
+  var numComChoice = 3;
+  var randomComChoice = generateNum(numComChoice);
+
+  if (randomComChoice == 1) {
```

Comment:
since you have 4 cases of randomComChoice here, it might be a good place to use switch statements. I know this is not covered in class but I thought it would be good to introduce you to it, so you can use the switch statement instead of if statements when you find yourself having multiple if statements of different case scenarios. Here, you might consider using a switch statement which could make your code cleaner and more readable. You can read more on it here https://www.w3schools.com/js/js_switch.asp